README.md

```diff
@@ -4,90 +4,62 @@ This is the Taskcluster team's internal terraform configuration for setting up
+*CAUTION*: that docker volume thus contains powerful cleartext secrets!
+Docker volumes are readable by anyone with permission to execute `docker run` on a host.
+**DO NOT RUN THIS TOOL ON A SHARED OR UNTRUSTED SYSTEM**
 
-Terraform must be built from source, along with a few providers.
+Once setup is complete, the script drops you in a shell at `/repo`.
+That's a bind mount of the repository where you ran `./terraform-runner.sh`.
+You can run `terraform` as much as you'd like in that docker container.
+You can also use the `kubectl`, `gcloud`, `az`, and `aws` tools from this environment to examine and administer the cluster.
 
-```shell
-mkdir go
-export GOPATH="$PWD/go"
-PATH="$PWD/go/bin:$PATH"
-go version   # needs to be go 1.10.x (don't pay attention to terraform repo's minimum version)
-go get github.com/hashicorp/terraform
-go get golang.org/x/tools/cmd/stringer
-( cd $GOPATH/src/github.com/hashicorp/terraform && make dev )
-```
+All other work (editing files, etc.) should occur outside of the docker container, as usual.
 
-And install the providers:
+### Existing Deployments
 
-```shell
-go get -u github.com/ericchiang/terraform-provider-k8s
-go get -u github.com/taskcluster/terraform-provider-jsone
-```
+If a deployment has already been set up, and you just want to modify it, change to the `install` directory.
+The first time around, you will need to run `terraform init` to install all of the various modules.
+Once that succeeds, `terraform plan` and `terraform apply` as usual.
 
-Note that in any new shells, you will need to set the PATH again:
-```shell
-PATH="$PWD/go/bin:$PATH"
-```
-
-## Usage
+### New Deployments
 
-On the first run of this only, you _must_ first create the kubernetes cluster by:
+If a deployment has not been set up, it will not have a Google Cloud configuration yet.
+In this case, run `terraform init` and `terraform apply` in the `setup` directory.
+Once that has completed successfully, edit `terraform-runner.sh` appropriately to add configuration for the new deployment and re-run it.
+Make a push to this repository to update the script with the new configuration.
 
-1. In the `setup` directory, you should `../go/bin/terraform init` and `../go/bin/terraform apply`. This will create
-a cluster and allow us to move onto the following steps.
+### Role Bindings
 
-If you have not yet configured your cluster in kubeconfig or you have set a different default
-cluster context, you can prepare for the following steps.
+I'm not sure what to do with this:
 
-2. `kubectl config get-contexts`
-3. `kubectl config use-context <cluster context>` -- select the context matching the gcloud cluster from the `get-contexts` list
-4. `kubectl create clusterrolebinding cluster-admin-binding-<your username> --clusterrole cluster-admin --user $(gcloud config get-value account)`
+`kubectl create clusterrolebinding cluster-admin-binding-<your username> --clusterrole cluster-admin --user $(gcloud config get-value account)`
```

Comment:
You just need to run this once after the cluster has been created. Maaaybe there is a way to do it with terraform directly but I gave up after a bit of trying.